src/oslog.cs

```diff
@@ -0,0 +1,164 @@
+	{
+		[Abstract]
+		[Export ("category")]
+		string Category { get; }
+
+		[Abstract]
+		[Export ("components")]
+		OSLogMessageComponent[] Components { get; }
+
+		[Abstract]
+		[Export ("formatString")]
+		string FormatString { get; }
+
+		[Abstract]
+		[Export ("subsystem")]
+		string Subsystem { get; }
+	}
+
+	[Mac (10,15)]
+	[BaseType (typeof(OSLogEntry))]
+	interface OSLogEntryActivity : OSLogEntryFromProcess
+	{
+		[Export ("parentActivityIdentifier")]
+		ulong ParentActivityIdentifier { get; }
+	}
+
+
+	[Mac (10,15)]
+	[BaseType (typeof(OSLogEntry))]
+	interface OSLogEntryLog : OSLogEntryFromProcess, OSLogEntryWithPayload
+	{
+		[Export ("level")]
+		LogLevel Level { get; }
+	}
+
+
+	[Mac (10,15)]
+	[BaseType (typeof(OSLogEntry))]
+	interface OSLogEntrySignpost : OSLogEntryFromProcess, OSLogEntryWithPayload
+	{
+		[Export ("signpostIdentifier")]
+		ulong SignpostIdentifier { get; }
+
+		[Export ("signpostName")]
+		string SignpostName { get; }
+
+		[Export ("signpostType")]
+		EntrySignpostType SignpostType { get; }
+	}
+
+
+	[Mac (10,15)]
+	[BaseType (typeof(NSObject))]
+	interface OSLogMessageComponent
+	{
+		[Export ("formatSubstring")]
+		string FormatSubstring { get; }
+
+		[Export ("placeholder")]
+		string Placeholder { get; }
+
+		[Export ("argumentCategory")]
+		ComponentArgumentCategory ArgumentCategory { get; }
+
+		[NullAllowed, Export ("argumentDataValue")]
+		NSData ArgumentDataValue { get; }
+
+		[Export ("argumentDoubleValue")]
+		double ArgumentDoubleValue { get; }
+
+		[Export ("argumentInt64Value")]
+		long ArgumentInt64Value { get; }
+
+		[NullAllowed, Export ("argumentNumberValue")]
+		NSNumber ArgumentNumberValue { get; }
+
+		[NullAllowed, Export ("argumentStringValue")]
+		string ArgumentStringValue { get; }
+
+		[Export ("argumentUInt64Value")]
+		ulong ArgumentUInt64Value { get; }
+	}
+
+	[Mac (10,15)]
+	[BaseType (typeof(NSObject))]
+	interface OSLogStore
+	{
+		[Static]
+		[Export ("localStoreAndReturnError:")]
+		[return: NullAllowed]
+		OSLogStore LocalStoreAndReturnError ([NullAllowed] out NSError error);
+
+		[Static]
+		[Export ("storeWithURL:error:")]
+		[return: NullAllowed]
+		OSLogStore StoreWithURL (NSUrl url, [NullAllowed] out NSError error);
+
+		[Export ("entriesEnumeratorWithOptions:position:predicate:error:")]
+		[return: NullAllowed]
+		NSEnumerator EntriesEnumeratorWithOptions (EnumeratorOptions options, [NullAllowed] NSObject position, [NullAllowed] NSPredicate predicate, [NullAllowed] out NSError error);
+
+		[Export ("entriesEnumeratorAndReturnError:")]
+		[return: NullAllowed]
+		NSEnumerator EntriesEnumeratorAndReturnError ([NullAllowed] out NSError error);
+
+		[Export ("positionWithDate:")]
+		NSObject PositionWithDate (NSDate date);
+
+		[Export ("positionWithTimeIntervalSinceEnd:")]
+		NSObject PositionWithTimeIntervalSinceEnd (double seconds);
+
+		[Export ("positionWithTimeIntervalSinceLatestBoot:")]
+		NSObject PositionWithTimeIntervalSinceLatestBoot (double seconds);
```

Comment:
Method must contain a verb:

```suggestion
		NSObject GetPositionWithTimeIntervalSinceLatestBoot (double seconds);
```

And shouldn't the return type be OSLogPosition?